x-pack/legacy/plugins/canvas/server/routes/public_workpad_snapshots.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { Server, Request, RouteOptions } from 'hapi';
+import { readFileSync } from 'fs';
+import { omit } from 'lodash';
+import { SavedObjectAttributes } from 'src/core/server';
+
+import { getId } from '../../public/lib/get_id';
+
+import {
+  SNAPSHOT_TYPE,
+  API_ROUTE_SNAPSHOT,
+  API_ROUTE_SNAPSHOT_RUNTIME,
+} from '../../common/lib/constants';
+
+// @ts-ignore
+import { RUNTIME_FILE } from '../../external_runtime/constants';
+import { CanvasRenderedWorkpad } from '../../external_runtime/types';
+
+const PUBLIC_OPTIONS: RouteOptions = {
+  auth: false,
+  cors: {
```

Comment:
@spalger @epixa I'm *sure* this isn't right.  I want to enable CORS for these public endpoints.  Can you please advise?

Reply:
I think it would make sense to expose this as config, disable cors initially, maybe stick some default in for dev, and remind people in the docs and in the UI that domains will be limited, how to change the config, etc. https://github.com/elastic/kibana/blob/3797fbf4eca9fd984f08cd230cd82c0a6a033e34/src/core/server/http/http_config.ts#L41-L55